sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -290,6 +291,22 @@ def optimizer(obj_func, initial_theta, bounds):
                        gpr.log_marginal_likelihood(gpr.kernel.theta))
 
 
+def test_gpr_correct_error_message():
+    import urllib2
+    import csv
+    f = urllib2.urlopen('http://pims.structuralbiology.eu/X.csv')
```

Comment:
Use some toy data rather than download the dataset in the original issue. Any matrix with a 0 eigenvalue should do I think, e.g. try:

```py
X = np.arange(12).reshape(6, -1)
y = np.ones(6)
gpr = GaussianProcessRegressor(kernel=DotProduct(), alpha=0)
gpr.fit(X, y)
```